x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformIndex.java

```diff
@@ -134,10 +134,28 @@ private static Settings createSettings() {
      * }
      * @param mappings A Map of the form {"fieldName": "fieldType"}
      */
-    private static Map<String, Object> createMappingsFromStringMap(Map<String, String> mappings) {
+    static Map<String, Object> createMappingsFromStringMap(Map<String, String> mappings) {
         Map<String, Object> fieldMappings = new HashMap<>();
-        mappings.forEach((k, v) -> fieldMappings.put(k, Map.of("type", v)));
-
+        for (Map.Entry<String, String> entry : mappings.entrySet()) {
+            String[] parts = entry.getKey().split("\\.");
+            Map<String, Object> current = fieldMappings;
+            current = diveInto(current, parts[0]);
+            for (int j = 1; j < parts.length; ++j) {
+                current = diveInto(current, "fields");
```

Comment:
Tests show this doesn't work when there are object fields mixed with multi-fields.
I need to rethink this solution.

Reply:
Ok, I've fixed it.
Still, I have the impression that there should be an easier way of reconciling individual mappings into the new index creation request. But that's what I have for now.